src/compiler/utils/string_with_sourcemap.ts

```diff
@@ -0,0 +1,269 @@
+import { DecodedSourceMap, RawSourceMap, SourceMapLoader } from '@ampproject/remapping/dist/types/types';
+import remapping from '@ampproject/remapping';
+import { SourceMap } from 'magic-string';
+
+type SourceLocation = {
+	line: number;
+	column: number;
+};
+
+function last_line_length(s: string) {
+	return s.length - s.lastIndexOf('\n') - 1;
+}
+
+// mutate map in-place
+export function sourcemap_add_offset(
+	map: DecodedSourceMap, offset: SourceLocation
+) {
+	if (map.mappings.length == 0) return map;
+	// shift columns in first line
+	const segment_list = map.mappings[0];
+	for (let segment = 0; segment < segment_list.length; segment++) {
+		const seg = segment_list[segment];
+		if (seg[3]) seg[3] += offset.column;
+	}
+	// shift lines
+	for (let line = 0; line < map.mappings.length; line++) {
+		const segment_list = map.mappings[line];
+		for (let segment = 0; segment < segment_list.length; segment++) {
+			const seg = segment_list[segment];
+			if (seg[2]) seg[2] += offset.line;
+		}
+	}
+}
+
+function merge_tables<T>(this_table: T[], other_table): [T[], number[], boolean, boolean] {
+	const new_table = this_table.slice();
+	const idx_map = [];
+	other_table = other_table || [];
+	let val_changed = false;
+	for (const [other_idx, other_val] of other_table.entries()) {
+		const this_idx = this_table.indexOf(other_val);
+		if (this_idx >= 0) {
+			idx_map[other_idx] = this_idx;
+		} else {
+			const new_idx = new_table.length;
+			new_table[new_idx] = other_val;
+			idx_map[other_idx] = new_idx;
+			val_changed = true;
+		}
+	}
+	let idx_changed = val_changed;
+	if (val_changed) {
+		if (idx_map.find((val, idx) => val != idx) === undefined) {
+			// idx_map is identity map [0, 1, 2, 3, 4, ....]
+			idx_changed = false;
+		}
+	}
+	return [new_table, idx_map, val_changed, idx_changed];
+}
+
+function pushArray<T>(_this: T[], other: T[]) {
+	for (let i = 0; i < other.length; i++)
+		_this.push(other[i]);
+}
+
+export class StringWithSourcemap {
+	string: string;
+	map: DecodedSourceMap;
+
+	constructor(string = '', map = null) {
+		this.string = string;
+		if (map)
```

Comment:
I think this will fail the new eslint config just checked in to `master`. Can you rebase against `master` and check that `npm run lint` passes? Btw, when fixing don't use the `--fix` option. It seems do so some pretty strange things for this rule